charts/htsget-rs/templates/ingress.yaml

```
@@ -0,0 +1,70 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "htsget-rs.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
```

```suggestion
apiVersion: networking.k8s.io/v1
```

Remember that you are targeting current stable k8s versions >= 1.26

---

In order not to exclude entirely the possibility of deploying the chart in older k8s clusters, we could release a first chart version keeping the  back-compatibility as is now and then move to `kubeVersion: "^1.26.0-0"` with a subsequent release.

---

Made changes to support >=1.19.0 k8s for now.